File: poetry-backend/src/main/java/com/poetry/poetry_backend/application/fingerprint/port/BatchDeleteResult.java
/*
 * File: BatchDeleteResult.java
 * Purpose: Result record for batch template deletion operations.
 * Contains counts and details of successful and failed slot deletions.
 * All Rights Reserved. Arodi Emmanuel
 */

package com.poetry.poetry_backend.application.fingerprint.port;

import java.util.List;

public record BatchDeleteResult(
        int deletedCount,
        List<Integer> failedSlots,
        boolean allSucceeded) {

    public static BatchDeleteResult success(int count) {
        return new BatchDeleteResult(count, List.of(), true);
    }

    public static BatchDeleteResult partial(int ok, List<Integer> failed) {
        return new BatchDeleteResult(ok, failed, false);
    }
}

File: poetry-hardware/src/application/ports/BatchDeleteResult.ts
// File: BatchDeleteResult.ts
// Purpose: Result type for batch template deletion operations.
// Contains successful and failed slot IDs with error details.
// All Rights Reserved. Arodi Emmanuel

export interface SlotError {
    readonly slotId: number;
    readonly error: string;
}

export interface BatchDeleteResult {
    readonly successfulSlots: number[];
    readonly failedSlots: SlotError[];
    readonly totalRequested: number;
}

export function createBatchResult(
    successful: number[],
    failed: SlotError[],
    total: number
): BatchDeleteResult {
    return {
        successfulSlots: successful,
        failedSlots: failed,
        totalRequested: total,
    };
}
